templates/plugin/src/index.ts

```diff
@@ -0,0 +1,108 @@
+import type { CollectionSlug, Config } from 'payload'
+
+export type MyPluginConfig = {
+  /**
+   * List of collections to add a custom field
+   */
+  collections?: Partial<Record<CollectionSlug, true>>
+  disabled?: boolean
+}
+
+export const myPlugin =
+  (pluginOptions: MyPluginConfig) =>
+  (config: Config): Config => {
+    if (pluginOptions.disabled) {
+      return config
+    }
+
+    if (!config.endpoints) {
+      config.endpoints = []
+    }
+
+    if (!config.admin) {
+      config.admin = {}
+    }
+
+    if (!config.admin.components) {
+      config.admin.components = {}
+    }
+
+    if (!config.admin.components.beforeDashboard) {
+      config.admin.components.beforeDashboard = []
+    }
+
+    config.admin.components.beforeDashboard.push(
+      `plugin-package-name-placeholder/client#BeforeDashboardClient`,
+    )
+    config.admin.components.beforeDashboard.push(
+      `plugin-package-name-placeholder/rsc#BeforeDashboardServer`,
+    )
+
+    if (!config.collections) {
+      config.collections = []
+    }
+
+    config.collections.push({
+      slug: 'plugin-collection',
+      fields: [
+        {
+          name: 'id',
+          type: 'text',
+        },
+      ],
+    })
+
+    if (pluginOptions.collections) {
+      for (const collectionSlug in pluginOptions.collections) {
+        const collection = config.collections.find(
+          (collection) => collection.slug === collectionSlug,
+        )
+
+        if (collection) {
+          collection.fields.push({
+            name: 'addedByPlugin',
+            type: 'text',
+            admin: {
+              position: 'sidebar',
+            },
+          })
+        }
+      }
+    }
+
+    config.endpoints.push({
+      handler: () => {
+        return Response.json({ message: 'Hello from custom endpoint' })
+      },
+      method: 'get',
+      path: '/my-plugin-endpoint',
+    })
+
+    const incomingOnInit = config.onInit
+
+    config.onInit = async (payload) => {
+      if (incomingOnInit) {
```

Comment:
Let's add a comment here about what this is doing, so it is not removed by mistake. Ex:

> Ensure we are executing any existing `onInit` functions.